### PR TITLE
allow setting values directly on slots; resolves #1397

### LINKED ISSUE #1397: Support for setting values directly on slots
related to #1208

### PR DESCRIPTION
**Pull request by kschzt**

### CODE REVIEW

File: browser/scripts/node.js [hunk 1]
```diff
@@ -239,6 +259,29 @@ Node.prototype.remove_slot = function(slot_type, suid) {
 	this.emit('slotRemoved', slot)
 }
 
+Node.prototype.setInputSlotValue = function(name, value) {
+	var slot = this.findInputSlotByName(name)
+
+	if (value === slot.def) {
+		delete this.uiSlotValues[name]
+		delete this.plugin.inputValues[name]
```
Comment (kkananen) [inline browser/scripts/node.js]: shouldn't inputValues still have a value for a slot even if it's the default?

File: browser/test/integration/uiSlotValues.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+var assert = require('assert');
```
Comment (kkananen) [inline browser/test/integration/uiSlotValues.js]: would be good to add tests for:
- update_input() gets called with default values before first update_state()

**Comment by kkananen** [comment]
Could we somehow protect against calling ChangeSlotValue when a slot is connected?

File: browser/scripts/node.js (hunk 1 above)
Comment (gmarinov) [inline browser/scripts/node.js]: i thought about that, but versioned engine already solves _most of_ that problem.

File: browser/test/integration/uiSlotValues.js (hunk 2 above)
Comment (kschzt) [inline browser/test/integration/uiSlotValues.js]: but it doesn't? this sets it to call update_input() with the overridden values on initialise()

**Comment by kschzt** [comment]
@kkananen now there is a test for that

**Comment by kkananen** [comment]
lgtm; would be good to test this for a while in develop before pushing to production

**Pull request merged**